c/planarityTestGraphFunctionality.c

```diff
@@ -408,15 +315,120 @@ int testAllGraphs(graphP theGraph, char command, char *inputStr, testAllStatsP s
 	if (freeG6ReadIterator(&pG6ReadIterator) != OK)
 		ErrorMessage("Unable to free G6ReadIterator.\n");
 	
-	return exitCode;
+	return Result;
 }
 
-int _getNumCharsToReprInt(int theNum)
+int outputTestAllGraphsResults(char command, testAllStatsP stats, char * infileName, char *outfileName, char **outputStr)
 {
-	int numCharsRequired = 1;
+	int Result = OK;
 
-	while(theNum /= 10)
-		numCharsRequired++;
+	FILE *outputFileP;
 
-	return numCharsRequired;
-}
\ No newline at end of file
+	int charsAvailForFilename = 0;
+	char *messageFormat = NULL;
+	char messageContents[MAXLINE + 1];
+
+	strOrFileP testOutput = NULL;
+
+	if (outfileName != NULL)
+	{
+		outputFileP = fopen(outfileName, "w");
+		if (outputFileP == NULL)
+		{
+			charsAvailForFilename = (int) (MAXLINE - strlen(outfileName));
+			messageFormat = "Unable to open file \"%.*s\" for output.\n";
+			sprintf(messageContents, messageFormat, charsAvailForFilename, outfileName);
+			ErrorMessage(messageContents);
+			return NOTOK;
+		}
+		
+		testOutput = sf_New(outputFileP, NULL);
+	}
+	else
+	{
+		if ((*outputStr) == NULL)
+			(*outputStr) = (char *) malloc(1 * sizeof(char));
+		
+		if ((*outputStr) == NULL)
+		{
+			ErrorMessage("Unable to allocate memory for outputStr.\n");
+			return NOTOK;
```

Comment:
Once you move the headerStr and resultStr mallocs up, you'd have to free them here, too.